testnet/stacks-node/src/nakamoto_node/miner.rs

```diff
@@ -1037,6 +1040,36 @@ impl BlockMinerThread {
         Some(vrf_proof)
     }
 
+    /// Wait the minimum time between blocks before mining a new block (if necessary)
+    /// This is to ensure that the signers do not reject the block due to the block being mined within the same second as the parent block.
+    fn wait_min_time_between_blocks(&self) -> Result<(), NakamotoNodeError> {
+        let burn_db_path = self.config.get_burn_db_file_path();
+        let mut burn_db =
+            SortitionDB::open(&burn_db_path, false, self.burnchain.pox_constants.clone())
+                .expect("FATAL: could not open sortition DB");
+
+        let mut chain_state = neon_node::open_chainstate_with_faults(&self.config)
+            .expect("FATAL: could not open chainstate DB");
+        let parent_block_info = self.load_block_parent_info(&mut burn_db, &mut chain_state)?;
```

Comment:
I'm guessing this works fine if this is the first Nakamoto block, where there is no parent block with a timestamp? I'm guessing tests would catch that.

Reply:
Yeah its the same logic used when building the block itself. No difference in the error path.